### PR TITLE
LPS-161386 Link To URL page with target doesn't open in new tab on page admin

### PR DESCRIPTION
**Pull request by ces-huynguyen**
LPS-161386

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by ces-huynguyen** [comment]
ci:test

**Comment by ruben-pulido** [comment]
ci:test:sf

**Comment by ruben-pulido** [comment]
ci:test:relevant

**Comment by victorg1991** [comment]
Hey @locpham97 @ces-huynguyen 

I'm afraid that this doesn't work properly, when configuring the target it adds a wrong targe to the link:
Screenshot 2022-08-30 at 13 25 09

Which may work but is not correct html

Apart from that, if you click in the title of the page it is not opening a new tab. Also when clicking in the pages tree in the product menu this behavior doesn't work either

Could you take a look?

Thanks!

**Comment by locpham97** [comment]
Hi @victorg1991 ,
I fix the target value displayed in HTML.
I'm not sure whether we should fix clicking on the title page or in the tree view. Could you verify it with team?

Thanks,
Loc

**Comment by victorg1991** [comment]
Yes @locpham97 we need to be consistent in all the places

**Comment by p2kmgcl** [comment]
image

@locpham97 I just checked (saw that you force pushed) and the target it is still wrong. Is that ready to be reviewed?

**Comment by locpham97** [comment]
Sorry @p2kmgcl , 
I will reupdate , thanks for reminding me

**Comment by locpham97** [comment]
Hi @p2kmgcl,
I just update this PR. Could you take a look at PR?
I am looking forward to hearing your response.
Thanks

File: modules/apps/product-navigation/product-navigation-product-menu-web/src/main/resources/META-INF/resources/portlet/pages_tree.jsp [hunk 1]
```diff
@@ -54,7 +54,7 @@ LayoutsTreeDisplayContext layoutsTreeDisplayContext = (LayoutsTreeDisplayContext
 				<clay:content-row
 					containerElement="span"
 				>
-					<a class="{cssClass}" data-regular-url="{regularURL}" data-url="{url}" data-uuid="{uuid}" href="{url}" id="{id}" title="{title}">
+					<a class="{cssClass}" data-regular-url="{regularURL}" data-url="{url}" data-uuid="{uuid}" href="{url}" id="{id}" target={target} title="{title}">
```
Comment (p2kmgcl) [inline modules/apps/product-navigation/product-navigation-product-menu-web/src/main/resources/META-INF/resources/portlet/pages_tree.jsp:57]: I think we are missing quotes here, shouldn't it be `target="{target}"`?

File: modules/apps/layout/layout-admin-web/src/main/resources/META-INF/resources/layout/advanced.jsp [hunk 2]
```diff
@@ -65,5 +65,12 @@ String targetType = GetterUtil.getString(layoutTypeSettings.getProperty("targetT
 
 <liferay-frontend:component
 	componentId='<%= liferayPortletResponse.getNamespace() + "addLayout" %>'
+	context='<%=
+		HashMapBuilder.<String, Object>put(
+			"defaultTarget", Objects.equals(targetType, "useNewTab") ? StringPool.BLANK : GetterUtil.getString(layoutTypeSettings.getProperty("target"))
```
Comment (p2kmgcl) [inline modules/apps/layout/layout-admin-web/src/main/resources/META-INF/resources/layout/advanced.jsp:70]: There are a couple things here that I am not sure about:

First, if `targetType` is `"useNewTab"`, why are we using `StringPool.BLANK` as target? Shouldn't it be `"_blank"` to open a new tab?

Second, I am not sure if we are using short if statements in Java (I never have seem them). I would try to refactor that into a `if` block creating a new `defaultTarget` variable. What do you think about this?

**Review by p2kmgcl — COMMENTED** [review]
Hey again, I have tried the PR and it works perfect :partying_face:. I also checked the code and I saw a couple of things, could you have a look at the comments, please?

File: modules/apps/layout/layout-admin-web/src/main/resources/META-INF/resources/layout/advanced.jsp (hunk 2 above)
Comment (locpham97) [reply modules/apps/layout/layout-admin-web/src/main/resources/META-INF/resources/layout/advanced.jsp:70]: The firstly, I set it StringPool.BLANK to make that when user change targetType from Use New Tag to Specific Frame, the target  will change to a empty string not being to a value "_blank".

Comment (p2kmgcl) [reply modules/apps/layout/layout-admin-web/src/main/resources/META-INF/resources/layout/advanced.jsp:70]: Looks perfect then. Thanks!

Comment (ealonso) [reply modules/apps/layout/layout-admin-web/src/main/resources/META-INF/resources/layout/advanced.jsp:70]: What about looking in google how to send good prs? In that way we are going to save a lot of time reviewing your pull many times. 

Think about it.

Regards,
Eudaldo.

Comment (jkappler) [reply modules/apps/layout/layout-admin-web/src/main/resources/META-INF/resources/layout/advanced.jsp:70]: @ces-huynguyen, friendly reminder for next time that we have Pull Request Templates: https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md

**Comment by ozysouza** [comment]
ci:forward

**Comment by ealonso** [comment]
ci:reevaluate:1287401_4426

**Comment by ealonso** [comment]
ci:forward

**Pull request closed without merging**